proxy/proxy.go

```diff
@@ -815,30 +815,37 @@ func (p *Proxy) checkRatelimit(ctx *context) (ratelimit.Settings, int) {
 		return ratelimit.Settings{}, 0
 	}
 
-	settings, ok := ctx.stateBag[ratelimitfilters.RouteSettingsKey].(ratelimit.Settings)
-	if !ok {
+	settings, ok := ctx.stateBag[ratelimitfilters.RouteSettingsKey].([]ratelimit.Settings)
+	if !ok || len(settings) < 1 {
 		return ratelimit.Settings{}, 0
 	}
 
-	rl := p.limiters.Get(settings)
-	if rl == nil {
-		return settings, 0
-	}
+	allow := true
+	for _, setting := range settings {
+		rl := p.limiters.Get(setting)
+		if rl == nil {
+			p.log.Errorf("RateLimiter is nil for setting: %s", setting)
+			continue
+		}
 
-	if settings.Lookuper == nil {
-		p.log.Error("lookuper is nil")
-		return settings, 0
-	}
-	s := settings.Lookuper.Lookup(ctx.Request())
+		if setting.Lookuper == nil {
+			p.log.Errorf("Lookuper is nil for setting: %s", setting)
+			continue
+		}
 
-	if s == "" {
-		return settings, 0
-	}
+		s := setting.Lookuper.Lookup(ctx.Request())
+		if s == "" {
+			p.log.Errorf("Lookuper found no data in request for setting: %s and request: %v", setting, ctx.Request())
+			continue
+		}
 
-	if rl.Allow(s) {
-		return settings, 0
+		allow = allow && rl.Allow(s)
```

Comment:
i think the `allow` var is not required. It could be just:

```
if !rl.Allow(s) {
    return setting, rl.RetryAfter(s)
}
```

Reply:
fixed